### DISCUSSION TITLE
give panel-tabsets an optional id so I can know when they open/close

### DISCUSSION THREAD

**Discussion by bryanwhiting** [quarto-dev/quarto-cli#5713] [Feature Requests]
### Description

The most annoying thing about panel-tabsets is figuring out where one starts and ends.

This is easy if you have two or three tabsets. But if you have a lot that are nested, you're suddenly left with code that looks like this:

```
# title
::: {panel-tabset}
## tab1
::: {panel-tabset}
### tab1.tab1
hi
### tab1.tab2
wo
:::
## tab2
something else
:::

```

Doesn't seem _that_ bad, but when you have a bunch of `:::{callout-tip}` life soon gets pretty miserable seeing `:::` everywhere and not knowing which `:::` ends which tabset.

Preferred behavior:
```
# title
::: {panel-tabset id=01}
## tab1
::: {panel-tabset id=02}
### tab1.tab1
hi
### tab1.tab2
wo
::: {end id=02}
## tab2
something else
::: {end id=01}
```

Obviously, others would know way better how to design this. But the thinking is if I know where it ends then I can more easily debug.

**Comment by mcanouil** [comment]
"id" has a meaning in HTML and this could be highly confusing.

See suggested syntax below.
Also, it's highly recommended to put headers on their own lines to avoid parsing issue (not only in Quarto).

```qmd

# title

::: {panel-tabset}
## tab1

:::: {panel-tabset}
### tab1.tab1

hi

### tab1.tab2

wo

::::
## tab2

something else
:::

```

Or

```qmd

# title

:::: {panel-tabset}
## tab1

::: {panel-tabset}
### tab1.tab1

hi

### tab1.tab2

wo

:::
## tab2

something else
::::

```

since this is a Pandoc design/feature and because I don’t think there is a need for a change, I am going to close this as a not planned.
Also, I don’t find the suggested complex syntax more readable.
